L1Trigger/TrackerTFP/src/SeedFilter.cc

```diff
@@ -0,0 +1,242 @@
+#include "L1Trigger/TrackerTFP/interface/SeedFilter.h"
+
+#include <numeric>
+#include <algorithm>
+#include <iterator>
+#include <deque>
+#include <vector>
+#include <set>
+#include <utility>
+#include <cmath>
+
+using namespace std;
+using namespace edm;
+using namespace trackerDTC;
+
+namespace trackerTFP {
+
+  SeedFilter::SeedFilter(const ParameterSet& iConfig, const Setup* setup, const DataFormats* dataFormats, const LayerEncoding* layerEncoding, int region) :
+    enableTruncation_(iConfig.getParameter<bool>("EnableTruncation")),
+    setup_(setup),
+    dataFormats_(dataFormats),
+    layerEncoding_(layerEncoding),
+    region_(region),
+    cot_(dataFormats_->format(Variable::cot, Process::sf)),
+    zT_(dataFormats_->format(Variable::zT, Process::sf)),
+    z_(dataFormats_->format(Variable::z, Process::sf)),
+    input_(dataFormats_->numChannel(Process::mht)) {}
+
+  void SeedFilter::consume(const TTDTC::Streams& streams) {
+    auto valid = [](int& sum, const TTDTC::Frame& frame){ return sum += (frame.first.isNonnull() ? 1 : 0); };
+    int nStubs(0);
+    for (int channel = 0; channel < dataFormats_->numChannel(Process::mht); channel++) {
+      const TTDTC::Stream& stream = streams[region_ * dataFormats_->numChannel(Process::mht) + channel];
+      nStubs += accumulate(stream.begin(), stream.end(), 0, valid);
+    }
+    stubsMHT_.reserve(nStubs);
+    for (int channel = 0; channel < dataFormats_->numChannel(Process::mht); channel++) {
+      const TTDTC::Stream& stream = streams[region_ * dataFormats_->numChannel(Process::mht) + channel];
+      vector<StubMHT*>& stubs = input_[channel];
+      stubs.reserve(stream.size());
+      for (const TTDTC::Frame& frame : stream) {
+        StubMHT* stub = nullptr;
+        if (frame.first.isNonnull()) {
+          stubsMHT_.emplace_back(frame, dataFormats_);
+          stub = &stubsMHT_.back();
+        }
+        stubs.push_back(stub);
+      }
+    }
+  }
+
+  void SeedFilter::produce(TTDTC::Streams& accepted, TTDTC::Streams& lost) {
+    auto kill = [this](const vector<StubSF*>& stubs) {
```

Comment:
These are very long for lambda functions, which are typically just a single line. If you want to keep them here, rather than upgrading the to full class member functions, please add a comment before them. "Define lambda functions" and another comment before code proper starts "End of lambda functions".